### PR TITLE
feat: add support for Sankey chart type

### LINKED ISSUE #5639: Chart with ChartType Sankey diagram
### Describe your motivation

We need a sankey diagram in our application and it would be great if the chart component would support this type of chart.

### Describe the solution you'd like

That the chart component support this chart type.

### Describe alternatives you've considered

Manually building a custom component for a sankey diagram with either highchart.js or D3.js

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by web-padawan**
## Description

Fixes #5639

Added `PlotOptionsSankey` based on https://api.highcharts.com/highcharts/series.sankey

## Type of change

- Feature

### CODE REVIEW

File: vaadin-charts-flow-parent/vaadin-charts-flow/src/main/java/com/vaadin/flow/component/charts/model/PlotOptionsSankey.java [hunk 1]
```diff
@@ -0,0 +1,767 @@
...
+    private Number opacity;
+    private Boolean selected;
+    private Boolean showCheckbox;
+    private Boolean showInLegend;
+    private Boolean skipKeyboardNavigation;
+    private States states;
+    private Boolean stickyTracking;
+    private Tooltip tooltip;
+    private Number turboThreshold;
+    private Boolean visible;
```
Comment (DiegoCardoso) [inline vaadin-charts-flow-parent/vaadin-charts-flow/src/main/java/com/vaadin/flow/component/charts/model/PlotOptionsSankey.java:57]: There are a few options that I can see from the Highcharts API that are missing here:

- allowPointSelect
- animation
- inactiveOtherPoints
- linkColorMode: from | to | gradient
- nodeAlignment: top | center | bottom

Comment (web-padawan) [reply vaadin-charts-flow-parent/vaadin-charts-flow/src/main/java/com/vaadin/flow/component/charts/model/PlotOptionsSankey.java:57]: > linkColorMode: from | to | gradient
> nodeAlignment: top | center | bottom

These two have been added in [`highcharts` 11.2.0 release](https://github.com/highcharts/highcharts/blob/6cc902cb843a20c05b6b895eaa0d82ee8dcbbb87/changelog/highcharts/11.2.0.md?plain=1#L5-L6) and we use an older version.
I'll check if other options should be added and update the PR.

Comment (DiegoCardoso) [reply vaadin-charts-flow-parent/vaadin-charts-flow/src/main/java/com/vaadin/flow/component/charts/model/PlotOptionsSankey.java:57]: Ah, right. I was relying on this information from their API page, but I guess it's not that reliable 🫠
image

Comment (web-padawan) [reply vaadin-charts-flow-parent/vaadin-charts-flow/src/main/java/com/vaadin/flow/component/charts/model/PlotOptionsSankey.java:57]: Yes, I also spent some time trying to figure out why `nodeAlignment` API didn't work 😕

Comment (web-padawan) [reply vaadin-charts-flow-parent/vaadin-charts-flow/src/main/java/com/vaadin/flow/component/charts/model/PlotOptionsSankey.java]: Added `allowPointSelect`, `animation` and `inactiveOtherPoints`.

**Review by DiegoCardoso — APPROVED** [review]

**Comment by netsrotr** [comment]
The new chart type seems to be not yet reflected in the current documentation at
[https://vaadin.com/docs/latest/components/charts/charttypes](https://vaadin.com/docs/latest/components/charts/charttypes)

**Pull request merged**